### DISCUSSION TITLE
is there any tag system in JCB?

### DISCUSSION THREAD

**Discussion by sabersaberi2** [joomengine/Joomla-Component-Builder#681] [Q&A]
hi.
I searched in wiki and tutorials but i can not find any topic for implement tag system in component according this :
https://docs.joomla.org/J3.x:Using_Tags_in_an_Extension
is JCB support tag system of joomla?
thank you

**Comment by Llewellynvdm** [comment]
Yes, JCB does have tags, you can just add the tag field to a view and it should work.
https://github.com/vdm-io/Joomla-Component-Builder/wiki/006-Basic-Fields#store-method-default-json-and-other-options
https://www.youtube.com/watch?v=9NO2rKnC6Ug&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&t=00h05m37s

To find answers in the tutorials of JCB you can use the Wiki, as it is word for word the transposed text of the tutorials. This makes it searchable and easy to find where what is explained. Like this:

https://github.com/vdm-io/Joomla-Component-Builder/search?q=tag+field&type=wikis

**Comment by sabersaberi2** [comment]
Thank you @Llewellynvdm 
i use that and deal some days with your tag field and joomla documentation about tag field. I understand that JCB missed implement some configuration of tag in joomla.
i want modify table and field mapping column params in content_types table. you can see what i say in this below link in the "admin/sql/install.mysql.utf8.sql" section:
https://docs.joomla.org/Special:MyLanguage/J3.2:Developing_an_MVC_Component/Adding_Tags
in the above link if i want use my own title in the 
"core_title": "greeting", 
how can i do that? if it is possible please advise me and describe correct solution about that. 
Thank you

**Reply by Llewellynvdm** [reply]
> I understand that JCB missed implement some configuration of tag in joomla.

I am not sure I follow, far as I know the tag field works correctly. But I don't use it that often, and I have not looked at any changes that may have happened in this area the last 4 years.

But listen the integration of this feature does do everything explained on https://docs.joomla.org/Special:MyLanguage/J3.2:Developing_an_MVC_Component/Adding_Tags automatically. So saying that JCB missed implement the tag field is not true. You will have to in the code show me the missed implementation, and if there has been a I change, we can just fix it. But just saying it missed implemented does not help. Honestly the hours I spend on making this part (tags,history,category integration) work correctly is immense. So if something is not working, I would really like to fix this... but in the components where I use tags, and stuff... those still work. So...

Now to have a `core_title` is not the same subject as the tags, though they are related in mapping. Yet I will answer it just quickly... by pointing that when you add a field to a view in JCB, there has a checkbox that you must check that says "Title":
![image](https://user-images.githubusercontent.com/5607939/110216847-0fdb9e00-7eba-11eb-9f48-e881a6dfb67d.png)

This is used to identify the `core_title` field name, and from that it maps it correctly. The one thing that will help you, is to believe that JCB does the impossible and more when you think it can't it is often because you just don't know how... This is partly my fault since I have not documented stuff beyond the basics... (too much to do) but if you can read PHP you will see much more under the hood of JCB than your average user.
